spdx/checksum.py

```diff
@@ -8,12 +8,49 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import re
+from enum import Enum, auto
+
+
+
+class ChecksumAlgorithmIdentifier(Enum):
+    SHA1 = auto()
+    SHA224 = auto()
+    SHA256 = auto()
+    SHA384 = auto()
+    SHA512 = auto()
+    SHA3_256 = auto()
+    SHA3_384 = auto()
+    SHA3_512 = auto()
+    BLAKE2B_256 = auto()
+    BLAKE2B_384 = auto()
+    BLAKE2B_512 = auto()
+    BLAKE3 = auto()
+    MD2 = auto()
+    MD4 = auto()
+    MD5 = auto()
+    MD6 = auto()
+    ADLER32 = auto()
+
+    def checksum_to_rdf(self):
+        return "checksumAlgorithm_" + self.name.lower()
+
+    @classmethod
+    def checksum_from_rdf(cls, identifier: str) -> str:
```

Comment:
The `match` here only refers to the regex which should catch any of the `BLAKE2b` algorithms as they need a special conversion, otherwise it is the default conversion to upper without the `checksumAlgorithm_`-prefix. The check, whether the identifier is valid (part of the enum), is done later when a new `Checksum` object is intialized.

Reply:
I refactored the `Checksum` class so that the identifier is a `ChecksumAlgorithm` and the enum class has a method to translate a `string` to a `ChecksumAlgorithm` that raises an error if the algorithm doesn't exist.